components/patient-data/impl/src/test/java/org/phenotips/data/internal/controller/GeneListControllerTest.java

```diff
@@ -0,0 +1,313 @@
+
+    @Test
+    public void loadCatchesExceptionFromDocumentAccess() throws Exception
+    {
+        Exception exception = new Exception();
+        doThrow(exception).when(this.documentAccessBridge).getDocument(any(DocumentReference.class));
+
+        PatientData<Map<String, String>> result = this.mocker.getComponentUnderTest().load(this.patient);
+
+        Assert.assertNull(result);
+    }
+
+    @Test
+    public void loadCatchesExceptionWhenPatientDoesNotHaveGeneClass() throws ComponentLookupException
+    {
+        doReturn(null).when(this.doc).getXObjects(any(EntityReference.class));
+
+        PatientData<Map<String, String>> result = this.mocker.getComponentUnderTest().load(this.patient);
+
+        Assert.assertNull(result);
+    }
+
+    @Test
+    public void loadAddsAllGenes() throws ComponentLookupException
+    {
+        PatientData<Map<String, String>> result = this.mocker.getComponentUnderTest().load(this.patient);
+
+        Assert.assertEquals(3, result.size());
+        Assert.assertTrue(result instanceof IndexedPatientData);
+        Assert.assertEquals("gene1", result.get(0).get(GENE_KEY));
+        Assert.assertEquals("comment1", result.get(0).get(COMMENTS_KEY));
+        Assert.assertEquals("gene2", result.get(1).get(GENE_KEY));
+        Assert.assertEquals("comment2", result.get(1).get(COMMENTS_KEY));
+        Assert.assertEquals("gene3", result.get(2).get(GENE_KEY));
+        Assert.assertEquals("comment3", result.get(2).get(COMMENTS_KEY));
+    }
+
+    //-----------------------------------writeJSON() tests-----------------------------------
+
+    @Test
+    public void writeJSONDoesNotWriteNullData() throws ComponentLookupException
+    {
+        doReturn(null).when(this.patient).getData(CONTROLLER_NAME);
+        JSONObject json = new JSONObject();
+        Collection<String> selectedFields = new LinkedList<>();
+        selectedFields.add(GENES_ENABLING_FIELD_NAME);
+        selectedFields.add(GENES_COMMENTS_ENABLING_FIELD_NAME);
+
+        this.mocker.getComponentUnderTest().writeJSON(this.patient, json, selectedFields);
+
+        Assert.assertNull(json.get(CONTROLLER_NAME));
+        verify(this.patient).getData(CONTROLLER_NAME);
+
+    }
+
+    @Test
+    public void writeJSONChecksThatSelectedFieldsContainsGeneEnabler() throws ComponentLookupException
+    {
+        List<Map<String, String>> internalList = new LinkedList<>();
+        PatientData<Map<String, String>> patientData = new IndexedPatientData<>(CONTROLLER_NAME, internalList);
+        doReturn(patientData).when(this.patient).getData(CONTROLLER_NAME);
+        JSONObject json = new JSONObject();
+        Collection<String> selectedFields = new LinkedList<>();
+        selectedFields.add("some_string");
+        selectedFields.add(GENES_COMMENTS_ENABLING_FIELD_NAME);
+
+        this.mocker.getComponentUnderTest().writeJSON(this.patient, json, selectedFields);
+
+        Assert.assertNull(json.get(CONTROLLER_NAME));
```

Comment:
Basically the test case is for when selectedFields does not contain `GeneListController.GENES_ENABLING_FIELD_NAME`. The behaviour being test for in that case is that the JSON is  not affected. If the current name makes that unclear, perhaps it could be renamed to: `writeJSONWhenSelectedFieldsDoesNotContainGeneEnabler`. Neither describes the actual behaviour particularly well but at least this describes the test case a little better?